### ISSUE TITLE
Variables in named variables

### ISSUE THREAD

**Issue by studiosciences** [less/less.js#2468]
I'd like to be able to use a variable for part of the variable name.

    @theme:Alternate;
    
    @themeColorDefault: blue;
    @themeColorAlternate: red;
    
    @themeColor: @@{themeColor@{theme}};  //    <--- WANT THIS    //
    
    body {
        color: @themeColor
    }

I realize I can do this by adding another variable, but it makes it really hard to follow especially when I have a lot of colors:

    @themeColorName:  "themeColorBase@{theme}";
    @themeColor:   @@themeColorName;

Thanks for your consideration.

Marc

**Comment by seven-phases-max** [comment]
Okay, since I face this kind of code quite often at the SO recently here's few comments on the use-case.

Basically the code above tries to emulate namespaces and #1848 in particular via variable identifier concatenation. So not condemning the feature itself (except its ugly syntax that propagates from string concatenation feature mostly by accident), I'm just remarking that the use-case is fundamentally unfortunate.

@studiosciences

I'd suggest you to rethink your theming code and consider to use namespace and mixins. Notice that while #1848 is not implemented yet you don't actually need it for this particular stuff, e.g.:

```
.theme(alternate) // using alternate theme here

.theme(alternate) {
    @color: blue;
}

.theme(default) {
    @color: red;
}

body {
    color: @color
}
```

(Obviously exact code may vary quite widely depending on actual needs, some more examples of using mixins as namespaces can be found in the mentioned #1848).

**Comment by studiosciences** [comment]
Thanks for your response, @seven-phases-max.

The problem stated in #1848 was variable isolation. It wasn't clear from the namespaces feature request whether the namespace itself could be a variable. If it could, that would 100% address my problem.

I hadn't quite realized I could use mixins to change a global variable, even though I had read through the scoping of mixin variables.
